### ISSUE TITLE
Can't build apk

### ISSUE THREAD

**Issue by sebastialonso** [inderps/react-redux-cordova-boilerplate#1]
I run the commands given in the README and this error I was given

```
BUILD SUCCESSFUL

Total time: 1 mins 44.941 secs
Built the following apk(s):
    /home/seba/repos/test_applications/react-redux-cordova-boilerplate/.cordova/platforms/android/build/outputs/apk/android-release-unsigned.apk
Please specify alias name

Please type jarsigner -help for usage
[11:38:02] 'build:android' errored after 2.62 min
[11:38:02] Error: Command `jarsigner -verbose -sigalg SHA1withRSA -digestalg SHA1 -keystore YOU-KEYSTORE-FILE.keystore ./platforms/android/build/outputs/apk/android-release-unsigned.apk` failed with exit code 1
    at ChildProcess.exithandler (child_process.js:744:12)
    at ChildProcess.emit (events.js:110:17)
    at maybeClose (child_process.js:1008:16)
    at Socket.<anonymous> (child_process.js:1176:11)
    at Socket.emit (events.js:107:17)
    at Pipe.close (net.js:476:12)

npm ERR! Linux 3.5.0-61-generic
npm ERR! argv "/home/seba/.nvm/v0.11.16/bin/node" "/home/seba/.nvm/v0.11.16/bin/npm" "run" "build:android"
npm ERR! node v0.11.16
npm ERR! npm  v2.3.0
npm ERR! code ELIFECYCLE
npm ERR! react-redux-cordova-boilerplate@1.0.0 build:android: `eslint scripts && gulp build:android`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the react-redux-cordova-boilerplate@1.0.0 build:android script 'eslint scripts && gulp build:android'.
npm ERR! This is most likely a problem with the react-redux-cordova-boilerplate package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     eslint scripts && gulp build:android
npm ERR! You can get their info via:
npm ERR!     npm owner ls react-redux-cordova-boilerplate
npm ERR! There is likely additional logging output above.
```

On dev server runs great. Thanks for the boilerplate! Any idea why it wouldn't work?

**Comment by inderps** [comment]
@sebastialonso Hey, I'm glad that you are finding this boilerplate useful.

To generate a apk file, you need to sign your application first.
As you can see the 'npm run build:android' task is written in [./scripts/tasks/android.js](https://github.com/inderps/react-redux-cordova-boilerplate/blob/master/scripts/tasks/android.js) file

There is this line which signs the application 

'jarsigner -verbose -sigalg SHA1withRSA -digestalg SHA1 -keystore YOU-KEYSTORE-FILE.keystore ./platforms/android/build/outputs/apk/android-release-unsigned.apk'

Basically you need to replace 'YOU-KEYSTORE-FILE.keystore' with your own keystore file path. Then It will run fine I guess

And to generate your own keystore file, you can refer this

http://developer.android.com/tools/publishing/app-signing.html#cert

Here you can go to 'Signing Your App Manually' section to see how to generate the keystore file.

Thanks and let me know if you face any other issues

**Comment by sebastialonso** [comment]
Thanks for the quick response! 

I replaced the segment with my own keystore file, but nothing different happened.

```
Please type jarsigner -help for usage
[12:06:07] 'build:android' errored after 28 s
[12:06:07] Error: Command `jarsigner -verbose -sigalg SHA1withRSA -digestalg SHA1 -keystore ~/keystore/android.jks ./platforms/android/build/outputs/apk/android-release-unsigned.apk` failed with exit code 1
    at ChildProcess.exithandler (child_process.js:744:12)
    at ChildProcess.emit (events.js:110:17)
    at maybeClose (child_process.js:1008:16)
    at Process.ChildProcess._handle.onexit (child_process.js:1080:5)
```

**Comment by inderps** [comment]
Can you post the full error log?

**Comment by sebastialonso** [comment]
Sure. Here it is 

```
...
:lintVitalRelease
:compileReleaseNdk
:preDexRelease
:dexRelease
:processReleaseJavaRes UP-TO-DATE
:packageRelease
:assembleRelease
:cdvBuildRelease

BUILD SUCCESSFUL

Total time: 17.373 secs
Built the following apk(s):
    /home/seba/repos/test_applications/react-redux-cordova-boilerplate/.cordova/platforms/android/build/outputs/apk/android-release-unsigned.apk
Please specify alias name

Please type jarsigner -help for usage
[12:18:43] 'build:android' errored after 24 s
[12:18:43] Error: Command `jarsigner -verbose -sigalg SHA1withRSA -digestalg SHA1 -keystore ~/keystore/android.jks ./platforms/android/build/outputs/apk/android-release-unsigned.apk` failed with exit code 1
    at ChildProcess.exithandler (child_process.js:744:12)
    at ChildProcess.emit (events.js:110:17)
    at maybeClose (child_process.js:1008:16)
    at Process.ChildProcess._handle.onexit (child_process.js:1080:5)

npm ERR! Linux 3.5.0-61-generic
npm ERR! argv "/home/seba/.nvm/v0.11.16/bin/node" "/home/seba/.nvm/v0.11.16/bin/npm" "run" "build:android"
npm ERR! node v0.11.16
npm ERR! npm  v2.3.0
npm ERR! code ELIFECYCLE
npm ERR! react-redux-cordova-boilerplate@1.0.0 build:android: `eslint scripts && gulp build:android`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the react-redux-cordova-boilerplate@1.0.0 build:android script 'eslint scripts && gulp build:android'.
npm ERR! This is most likely a problem with the react-redux-cordova-boilerplate package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     eslint scripts && gulp build:android
npm ERR! You can get their info via:
npm ERR!     npm owner ls react-redux-cordova-boilerplate
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/seba/repos/test_applications/react-redux-cordova-boilerplate/npm-debug.log
```

Tell me if there's anything else I can do. Thanks!

**Comment by inderps** [comment]
@sebastialonso As the error log says 'Please specify alias name'

you also have to specify the alias and storepass of your keystore

the new command would be

jarsigner -verbose -sigalg SHA1withRSA -digestalg SHA1 -keystore ~/keystore/android.jks ./platforms/android/build/outputs/apk/android-release-unsigned.apk  "ALIAS NAME" -storepass STOREPASS

**Comment by sebastialonso** [comment]
There's progress! The .jar has been signed!

A new error comes up

```
  signing: res/drawable-land-mdpi-v4/screen.png
  signing: res/drawable-land-xhdpi-v4/screen.png
  signing: res/drawable-ldpi-v4/icon.png
  signing: res/drawable-mdpi-v4/icon.png
  signing: res/drawable-port-hdpi-v4/screen.png
  signing: res/drawable-port-ldpi-v4/screen.png
  signing: res/drawable-port-mdpi-v4/screen.png
  signing: res/drawable-port-xhdpi-v4/screen.png
  signing: res/drawable-xhdpi-v4/icon.png
  signing: res/xml/config.xml
  signing: resources.arsc
  signing: classes.dex
jar signed.

Warning: 
No -tsa or -tsacert is provided and this jar is not timestamped. Without a timestamp, users may not be able to validate this jar after the signer certificate's expiration date (2040-12-03) or after any future revocation date.
jar verified.

Warning: 
This jar contains entries whose certificate chain is not validated.
This jar contains signatures that does not include a timestamp. Without a timestamp, users may not be able to validate this jar after the signer certificate's expiration date (2040-12-03) or after any future revocation date.

Re-run with the -verbose and -certs options for more details.
/bin/sh: 1: zipalign: not found
[12:44:10] 'build:android' errored after 60 s
[12:44:10] Error: Command `zipalign -v 4 ./platforms/android/build/outputs/apk/android-release-unsigned.apk LebeBalance.apk` failed with exit code 127
    at ChildProcess.exithandler (child_process.js:744:12)
    at ChildProcess.emit (events.js:110:17)
    at maybeClose (child_process.js:1008:16)
    at Socket.<anonymous> (child_process.js:1176:11)
    at Socket.emit (events.js:107:17)
    at Pipe.close (net.js:476:12)
```

Is zipalign a shell command?

Edit: It turns out I do have zipalign :(

```
-> % locate zipalign
/home/seba/Android/Sdk/build-tools/21.1.2/zipalign
/home/seba/Android/Sdk/build-tools/22.0.0/zipalign
/home/seba/Android/Sdk/build-tools/22.0.1/zipalign
/home/seba/Android/Sdk/build-tools/23.0.2/zipalign
/home/seba/Android/Sdk/docs/tools/help/zipalign.html
```

**Comment by inderps** [comment]
yeah it should be available at global level or change the script to use '/home/seba/Android/Sdk/build-tools/21.1.2/zipalign' instead of 'zipalign'

Also you need to specify the name of app in config.js

after this task is successful, you can find the apk inside .cordova directory

**Comment by sebastialonso** [comment]
Thank you very much @inderps. It works great! 

Thanks for your patience!
